src/main/java/com/coreos/jetcd/internal/impl/MaintenanceImpl.java

```diff
@@ -133,65 +97,27 @@
    */
   @Override
   public CompletableFuture<DefragmentResponse> defragmentMember(String endpoint) {
-    Optional<Pair<ManagedChannel, Optional<String>>> pairOptional = Optional.empty();
-    try {
-      pairOptional = Optional.of(this.dialFunction.dial(endpoint));
-      Pair<ManagedChannel, Optional<String>> pair = pairOptional.get();
-      ManagedChannel channel = pair.getKey();
-      Optional<String> token = pair.getValue();
-      MaintenanceGrpc.MaintenanceFutureStub stub = ClientUtil
-          .configureStub(MaintenanceGrpc.newFutureStub(channel), token);
-      ListenableFuture<DefragmentResponse> defragmentResponseListenableFuture = stub
-          .defragment(DefragmentRequest.getDefaultInstance());
-
-      // close channel when defragmentResponseListenableFuture completes.
-      defragmentResponseListenableFuture
-          .addListener(() -> channel.shutdownNow(), this.executorService);
-      return FutureConverter.toCompletableFuture(defragmentResponseListenableFuture);
-    } catch (Exception e) {
-      pairOptional.ifPresent((pair) -> pair.getKey().shutdownNow());
-      throw new RuntimeException("defragmentMember encounters error", e.getCause());
-    }
+    return toCompletableFuture(this.stub.defragment(DefragmentRequest.getDefaultInstance()));
   }
 
   /**
    * get the status of one member.
    */
   @Override
   public CompletableFuture<StatusResponse> statusMember(String endpoint) {
-    Optional<Pair<ManagedChannel, Optional<String>>> pairOptional = Optional.empty();
-    try {
-      pairOptional = Optional.of(this.dialFunction.dial(endpoint));
-      Pair<ManagedChannel, Optional<String>> pair = pairOptional.get();
-      ManagedChannel channel = pair.getKey();
-      Optional<String> token = pair.getValue();
-      MaintenanceGrpc.MaintenanceFutureStub stub = ClientUtil
-          .configureStub(MaintenanceGrpc.newFutureStub(channel), token);
-      ListenableFuture<StatusResponse> statusResponseListenableFuture = stub
-          .status(StatusRequest.getDefaultInstance());
-
-      // close channel when statusResponseListenableFuture completes.
-      statusResponseListenableFuture
-          .addListener(() -> channel.shutdownNow(), this.executorService);
-      return FutureConverter.toCompletableFuture(statusResponseListenableFuture);
-    } catch (Exception e) {
-      pairOptional.ifPresent((pair) -> pair.getKey().shutdownNow());
-      throw new RuntimeException("statusMember encounters error", e.getCause());
-    }
+    return toCompletableFuture(this.stub.status(StatusRequest.getDefaultInstance()));
```

Comment:
`statusMember ` takes an endpoint. The request must send to that specific endpoint in order for this call to be correct.